### DISCUSSION TITLE
ExternalPayloadStorage (S3Payload)

### DISCUSSION THREAD

**Discussion by vamsibandarupalli** [Netflix/conductor#2287] [Q&A] [answered]
Hey, I'm using S3PayloadStorage and it's working great, but how can we clean up the s3bucket objects after completion of workflow?

Since objects in s3 bucket are storing with the name of taskId or workflowId they won't be useful after workflow completion. 

Is there a way we can clean s3 objects after completion of workflow?

**Comment by apanicker-nflx** [comment, marked as answer]
Hi @vamsibandarupalli Currently, we use a TTL on our S3 buckets that automatically deletes the objects. This TTL is chosen to be large enough so that it is sufficiently higher than workflow execution time. 
Alternatively, you could customize your implementation of the WorkflowStatusListener to clean up the specific S3 objects that pertain to your workflow upon completion.
